docs/USAGE-GUIDE.md

```diff
@@ -1155,15 +1155,22 @@ file-descriptor should be active and connected. s2n-tls also supports setting th
 read and write file-descriptors to different values (for pipes or other unusual
 types of I/O).
 
+Important Note:
+Applications should be designed so that one process is reading from a pipe before another
+writes to it. If the reading process fails to start or is terminated unexpectedly, writing
```

Comment:
This could be considered a repeat of "This file-descriptor should be active and connected" above. Maybe replace your first line with that instead?

I'd also bold / underline / do something to make "Important Note" stand out.

Reply:
do you mean delete “Applications should be designed..” or combine them?

The “Applications should be designed..” came from gnu documentation and it seemed more clear even for those not familiar with C development. I would prefer to combine or remove "This
file-descriptor should be active"